feature/manager.go

```diff
@@ -235,7 +235,7 @@ func (m *Manager) UpdateFeatureSets(
 	// passed validation so that we don't partially update any sets then
 	// fail out on a later set's validation.
 	for set, features := range updates {
-		m.SetRaw(set, features.Clone())
+		m.setRaw(set, features.Clone())
```

Comment:
Q: Wondering why we need to clone the feature set here?

Reply:
Maps are [pointer types](https://dave.cheney.net/2017/04/30/if-a-map-isnt-a-reference-variable-what-is-it), so we need to clone the features so that the caller can't accidentally mutate what's put in our features map. 

https://go.dev/play/p/yMALps5fbh6

Reply:
Ok sounds good, thought because its a setter anyways, it does not change them (or should not) but I agree better play it safe here. Thanks for the coding example